theano/gof/destroyhandler.py

```diff
@@ -394,6 +394,17 @@ def get_destroyers_of(r):
                 return []
         fgraph.destroyers = get_destroyers_of
 
+        def has_destroyers(l):
+            for item in l:
+                droot, _, root_destroyer = self.refresh_droot_impact()
+                try:
+                    [root_destroyer[droot[item]]]
+                    return True
+                except Exception:
```

Comment:
I know you copied the Exception from above, but this is a bad practice. We should try to catch a much more precise exception.

Anyway, this isn't the optimized algorythm anyway. Both otherwise, you seem to have done the first part well. If with my modif in the other comment and tests pass, the structural change would be done. It would only miss the optimized implementation here.